cmd/src/users_clean.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/sourcegraph/src-cli/internal/api"
+)
+
+func init() {
+	usage := `
+Examples:
+
+	$ src users clean -d 182
+
+`
+
+	flagSet := flag.NewFlagSet("clean", flag.ExitOnError)
+	usageFunc := func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "Usage of 'src users %s':\n", flagSet.Name())
+		flagSet.PrintDefaults()
+		fmt.Println(usage)
+	}
+	var (
+		daysToDelete = flagSet.Int("d", 365, "Day threshold on which to remove users, defaults to 365")
+		noAdmin      = flagSet.Bool("no-admin", false, "Omit admin accounts from cleanup")
+		// TODO: Add an email functionality to email removed users, open editor like in git commit
+		// toEmail            = flagSet.Bool("email", false, "send removed users an email")
+		removeNoLastActive = flagSet.Bool("removeNeverActive", false, "removes users with null lastActive value")
+		skipConfirmation   = flagSet.Bool("skip-conf", false, "skips user confirmation step allowing programmatic use")
+		// TODO: Write json file containing users to be deleted, last active usage, site-admin status, and emails
+		// json               = flagSet.Bool("json", false, "Write json file containing users to be deleted, last active usage, site-admin status, and emails")
+		apiFlags = api.NewFlags(flagSet)
+	)
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			fmt.Printf("failed to parse command args with error: %s", err)
+		}
+
+		ctx := context.Background()
+		client := cfg.apiClient(apiFlags, flagSet.Output())
+
+		vars := map[string]interface{}{
+			"-d": api.NullInt(*daysToDelete),
+		}
+
+		query := `
+query Users($first: Int, $query: String) {
+	users(first: $first, query: $query) {
+		nodes {
+			...UserFields
+		}
+	}
+}
+` + userFragment
+
+		// get users to delete
+		var result struct {
+			Users struct {
+				Nodes []User
+			}
+		}
+		if ok, err := client.NewRequest(query, vars).Do(ctx, &result); err != nil || !ok {
+			return err
+		}
+
+		usersToDelete := make([]UserToDelete, 0)
+		for _, user := range result.Users.Nodes {
+			daysSinceLastUse, wasLastActive, err := computeDaysSinceLastUse(user)
+			if err != nil {
+				return err
+			}
+			if !wasLastActive && !*removeNoLastActive {
+				continue
+			}
+			if *noAdmin && user.SiteAdmin {
+				continue
+			}
+			if daysSinceLastUse <= *daysToDelete && wasLastActive {
+				continue
+			}
+			deleteUser := UserToDelete{user, daysSinceLastUse}
+
+			usersToDelete = append(usersToDelete, deleteUser)
+			//fmt.Printf("\nAdding %s to remove list: %d days since last active, remove after %d days inactive\n", user.Username, daysSinceLastUse, *daysToDelete)
```

Comment:
Do we want to keep this line? If not we can just delete it.

Reply:
ty missed it 👍